utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -239,11 +239,11 @@ impl PalletCmd {
 					let diff = highest - lowest;
 
 					// Create up to `STEPS` steps for that component between high and low.
-					let step_size = (diff / self.steps).max(1);
-					let num_of_steps = diff / step_size + 1;
-					for s in 0..num_of_steps {
+					let step_size = (diff as f32 / (self.steps - 1) as f32).max(1.0);
```

Comment:
Could you still please handle the edge-cases?  
Does this just panic when called with `steps <= 1`?  
In the `1` case the step size is irrelevant anyway.

Reply:
when `steps == 1` it does panic, but it's not because of a division by `0`.
doing the following did not solve the issue:
```
let step_size = if self.steps == 1 {
    1.0
} else {
    (diff as f32 / (self.steps - 1) as f32).max(1.0)
};
```

actually, it seems that the logic under [`median_slopes`](https://github.com/paritytech/substrate/blob/dbdbef95c84644328bee60bd170d79814c04e32b/frame/benchmarking/src/analysis.rs#L106) function isn't prepared to handle such edge case.

```
$ cargo run --features runtime-benchmarks -- benchmark pallet --pallet pallet_utility --extrinsic batch --steps 1 --low 0 --high 50
   Compiling frame-benchmarking-cli v4.0.0-dev (/home/bear/develop/substrate/utils/frame/benchmarking-cli)
   Compiling node-cli v3.0.0-dev (/home/bear/develop/substrate/bin/node/cli)
    Finished dev [unoptimized + debuginfo] target(s) in 4m 17s
     Running `target/debug/substrate benchmark pallet --pallet pallet_utility --extrinsic batch --steps 1 --low 0 --high 50`
2022-07-22 17:40:03 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
Pallet: "pallet_utility", Extrinsic: "batch", Lowest values: [0], Highest values: [50], Steps: 1, Repeat: 1
Raw Storage Info
========

Median Slopes Analysis
========

====================

Version: 3.0.0-dev-b9a764afbf3

   0: sp_panic_handler::panic_hook
             at primitives/panic-handler/src/lib.rs:166:18
...
  19: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/b17226fcc11587fed612631be372a5b4cb89988a/library/alloc/src/vec/mod.rs:2547:9
  20: core::iter::traits::iterator::Iterator::collect
             at /rustc/b17226fcc11587fed612631be372a5b4cb89988a/library/core/src/iter/traits/iterator.rs:1778:9
  21: frame_benchmarking::analysis::Analysis::median_slopes
             at frame/benchmarking/src/analysis.rs:151:16
  22: frame_benchmarking_cli::pallet::command::<impl frame_benchmarking_cli::pallet::PalletCmd>::print_summary
             at utils/frame/benchmarking-cli/src/pallet/command.rs:460:6
  23: frame_benchmarking_cli::pallet::command::<impl frame_benchmarking_cli::pallet::PalletCmd>::run
             at utils/frame/benchmarking-cli/src/pallet/command.rs:399:4
  24: node_cli::command::run::{{closure}}
             at bin/node/cli/src/command.rs:114:7
...
```
would someone ever want to benchmark with a single step? is it worth investigating on the necessary adaptations to `median_slopes` logic?


for now I'm just adding a check that avoids `steps <= 1`.

Reply:
Okay I see. Its because the slope logic needs at least two points to work.  
I will push a fix that should make it work together.   
A small comparison of the `sr25519_verification` benchmark which formerly overshot a lot:

```pre
...
Running Benchmark: frame_benchmarking.sr25519_verification 99/50 1/1
Running Benchmark: frame_benchmarking.sr25519_verification 100/50 1/1
```

Now its on point. This should also reduce the runtime of the benchmarks.
```pre
... 
Running Benchmark: frame_benchmarking.sr25519_verification 46/50 1/1
Running Benchmark: frame_benchmarking.sr25519_verification 48/50 1/1
```

Reply:
yes, reduction of benchmarks will be great!